src/VisualStudio/Core/Def/Implementation/LanguageClient/AlwaysActivateInProcLanguageClient.cs

```diff
@@ -17,19 +17,21 @@
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
 {
     /// <summary>
-    /// Language client responsible for exposing pull diagnostics.  Expected to run in all scenarios where the host
-    /// supports pull diagnostics (including VS and liveshare).
+    /// Language client responsible for handling C# / VB LSP requests in any scenario (both local and codespaces).
+    /// This powers "LSP only" features (e.g. cntrl+Q code search) that do not use traditional editor APIs.
+    /// It is always activated whenever roslyn is activated.
     /// </summary>
     [ContentType(ContentTypeNames.CSharpContentType)]
     [ContentType(ContentTypeNames.VisualBasicContentType)]
     [Export(typeof(ILanguageClient))]
-    internal class PullDiagnosticsInProcLanguageClient : AbstractInProcLanguageClient
+    [Export(typeof(AlwaysActivateInProcLanguageClient))]
+    internal class AlwaysActivateInProcLanguageClient : AbstractInProcLanguageClient
```

Comment:
This is active in live share and local, right? So what happens with workspace symbol requests given that both servers support them? Do we need this to change capabilities if live share is active, or will the platform handle that (ie, it won't send request to this one if live share is active)?

I guess this is a long winded way of asking if this is really `LocalInProcLanguageClient`?

Reply:
this provides cntrl+Q search for codespaces as well, while the liveshare server provides cntrl+, in codespaces.  This isn't ideal and is complicated, so I'm working with Christian to have codespaces only use this one (and remove the capability from the liveshare one)